generics/services/users.js

```diff
@@ -235,9 +235,73 @@ const fetchPublicTenantDetails = function (tenantId) {
 }
 
 
+/**
+ * Fetches user profile by userId/username and tenantId.
+ * @param {String} tenantId - tenantId details
+ * @param {String} userId - userId details
+ * @param {String} username - username details
+ * @returns {Promise} A promise that resolves with the organization details or rejects with an error.
+ */
+
+const fetchProfileBasedOnUserIdOrName = function (tenantId,userId=null,username) {
+	return new Promise(async (resolve, reject) => {
+		try {
+
+			let params;
+			if(userId){
```

Comment:
@borkarsaish65 Please correct the alignment and indentation, and format the code properly across your files